engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -964,7 +965,13 @@ else if (allocatedContainers.isEmpty() && numRequestedContainers == 0 && dnmgr.c
         appDone = true;
       }
 
-      LOG.debug("Current application state: loop=" + loopCounter + ", appDone=" + appDone + ", total=" + numTotalContainers + ", requested=" + numRequestedContainers + ", released=" + numReleasedContainers + ", completed=" + numCompletedContainers + ", failed=" + numFailedContainers + ", currentAllocated=" + allocatedContainers.size());
+      if (!StringUtils.isEmpty(dnmgr.shutdownDiagnosticsMessage) && !appDone) {
```

Comment:
Just for readability - as per the feedback from @sandeshh . I can combine this into a single statement but I don't think it really makes a difference one way or the other.